devito/symbolics/extended_sympy.py

```diff
@@ -374,6 +387,17 @@ class Cast(UnaryOp):
     __rkwargs__ = ('stars',)
 
     def __new__(cls, base, stars=None, **kwargs):
+        # Attempt simplifcation
+        # E.g., `FLOAT(32) -> 32.0` of type `sympy.Float`
+        try:
+            return sympify(eval(cls._base_typ)(base))
```

Comment:
THis will loose accuracy for `double` but not sure what's the easier way as there is only `float` and `int` really defined as python types

Reply:
no unless there's an 'F' in C all numbers with digits are `double`, so we should be good